RotatedSortedArraySearch.java

```diff
@@ -0,0 +1,39 @@
+// Time Complexity : log(n)
+// Space Complexity : log(n)
+// Did this code successfully run on Leetcode : yes
+// Any problem you faced while coding this : (confuse in space complexity)
```

Comment:
You are not using any extra space here i.e. you are not using any auxiliary data structure. So it will be constant